Pull-request для того, чтобы что-то было, если вдруг что пойдет не так перед рк.

KirSlayk opened the pull request:
ViewManager
Logout
Session
Change Router

Commit:
Добавлены либы для моков, каркас модели авторизации

Commit:
merge with v2

Commit:
use sinon js for mocks

DmitryDorofeev committed:
merge with v2

DmitryDorofeev committed:
fix jar

Commit:
api proxy to apiary

KirSlayk committed:
commit before rk

KirSlayk committed:
merge solved

KirSlayk committed:
deleted bootstrap.js from js dir

FakeEmperor committed:
Major update:
[+] Added unit tests for models/score
[+] Added Session model skeleton
[*] Refactored and fixed some errors in server.js
[+] Added livereload ability
[*] Added SCSS support and remade login.css style (it's now login.scss)
[*] Bugfix in password pattern in login.xml template
[*] Changed login block name in login.xml template

KirSlayk committed:
some change

KirSlayk committed:
pull-request was accepted

KirSlayk committed:
Merge branch 'pineapple_working'

KirSlayk committed:
logout function

KirSlayk committed:
logout function

KirSlayk committed:
Merge branch 'slaykV2'

KirSlayk committed:
ViewManager added

KirSlayk committed:
Merge branch 'slaykV2'

KirSlayk committed:
Session changed, more functions

KirSlayk committed:
Merge branch 'slaykV2'

DmitryDorofeev committed:
use travis

DmitryDorofeev commented:
вмержте себе ветку v2 из frontend-upstream, чтобы запускались тесты в travis-ci

Commit:
:Merge branch 'v2' of github.com:frontend-park-mail-ru/frontend-upstream into v3

KirSlayk commented:
с v2 мы уже up-to-date давно. Мб надо v3?

KirSlayk commented:
@DmitryDorofeev с v2 мы уже up-to-date давно. Мб надо v3?

DmitryDorofeev commented:
Уже не up-to-date. Можете и v3 мержить, без разницы.

KirSlayk committed:
update to upstream/v3

KirSlayk committed:
added top-gif and button's style

KirSlayk committed:
session model update - save, fetch, destroy. Not postAuth()

KirSlayk committed:
users_manager with session

FakeEmperor committed:
Design commit.
[+] Main menu design at /Design
[+] Card design and frames

FakeEmperor committed:
Design commit.
[+] Main menu design at /Design
[+] Card design and frames

FakeEmperor committed:
Merge branch 'pineapple_working' of https://github.com/SlaykovskiyHDNANO/2016_1_Lime-n-Pineapple into pineapple_working

KirSlayk committed:
Merge pull request #4 from SlaykovskiyHDNANO/pineapple_working

Pineapple working

KirSlayk committed:
conflict

KirSlayk committed:
Merge branch 'master' of https://github.com/SlaykovskiyHDNANO/2016_1_Lime-n-Pineapple

KirSlayk committed:
changed design of menu (pineapple's design)

KirSlayk committed:
dising

KirSlayk committed:
some change

KirSlayk committed:
.travis.yml upstream/v2

KirSlayk committed:
upstream/v3

KirSlayk committed:
tests complete

KirSlayk commented:
@DmitryDorofeev  примите, пожалуйста, pull-request

KirSlayk commented:
@madimp примите pull-request, пожалуйста

madimp commented on public_html/css/login.scss:
```diff
@@ -0,0 +1,23 @@
+
+
+.js-form__login {
+  margin-top: 5px;
```
классы с префиксами js нужны для использования в скриптах, а не стилях

madimp commented on public_html/js/models/session.js:
```diff
@@ -0,0 +1,104 @@
+                    success: function(model, res){
+                        console.log("SUCCESS");
+                        this.user.set({logged_in: true, login: opts.login});
+                        $("#login").text("Logout");
```
в модели не должно быть ничего про верстку

madimp commented on public_html/css/login.css:
```diff
@@ -1,5 +1,39 @@
+
+}
+
+#btn-login
```
id совсем не бэм

madimp commented on public_html/css/main.css:
```diff
@@ -1,5 +1,131 @@
+}
+
+
+#view__top
```
id совсем не бэм

madimp commented on public_html/css/main.css:
```diff
@@ -1,5 +1,131 @@
+    /*align-content: center;*/
+}
+
+#menu {
```
id совсем не бэм

madimp commented on public_html/index.html:
```diff
@@ -1,27 +1,21 @@
 	    <title>Whack-a-mole: Technopark Scholar Edition</title>
-
+		<link rel="stylesheet" href="css/main.css" />
+		<link rel="stylesheet" href="css/bootstrap.css" />
```
сначала надо подключать библиотеки, потом свои стили

madimp commented on public_html/js/models/card.js:
```diff
@@ -0,0 +1,18 @@
+                name: "",
+                img: ""
+            },
+            initialize: function (model) {
```
отступы

madimp commented on public_html/js/models/card.js:
```diff
@@ -0,0 +1,18 @@
+            },
+            initialize: function (model) {
+                console.log("Card initialize");
+                this.img = model.img;
```
модель инициализировать моделью? в чем прикол?

madimp commented on public_html/js/models/session.js:
```diff
@@ -0,0 +1,104 @@
+            },
+
+            initialize: function () {
+                UsersManager.attachEvent(this);
```
модель использует вью? очень странно

madimp commented on public_html/js/models/session.js:
```diff
@@ -0,0 +1,104 @@
+                this.user.save({login: opts.login, password: opts.password, logged_in: true}, {
+                    success: function(model, res){
+                        console.log("SUCCESS");
+                        this.user.set({logged_in: true, login: opts.login});
```
угадайте, какой тут будет this?

madimp commented on public_html/js/models/session.js:
```diff
@@ -0,0 +1,104 @@
+                        Backbone.history.history.back();
+                    },
+                    error: function(model, res){
+                        Backbone.history.history.back();
```
куда и вообще почему нужно перейти при ошибке разлогина?

KirSlayk replied:
на предыдущую страницу, которая была перед login

KirSlayk replied:
в error это пока что присутствует, потому что нет связки с сервером. Потом уберу

madimp replied:
если я вызвал Session.logout и произошла ошибка, почему меня должно на какую-то там страницу перекинуть?

madimp commented on public_html/js/models/session.js:
```diff
@@ -0,0 +1,104 @@
+                        this.user.set({logged_in: false, login: ""});
+                        $("#login").text("Login");
+                        $("#login").attr('href', "#login");
+                        Backbone.history.history.back();
```
куда и вообще почему нужно перейти при разлогине?

KirSlayk replied:
нет связки с сервером, потом поменяю поведение

madimp commented on public_html/js/models/user.js:
```diff
@@ -0,0 +1,21 @@
+            },
+            initialize: function () {
+                console.log("[ScoreModel::initialize]", "initalizing...");
+                this.on("increment", function (value) {
```
что за событие increment?
кто-то сбоку будет его тригерить?
события нужны не для действий с объектом, а для реакции внешних компонентов на действия объекта

madimp commented on public_html/js/router.js:
```diff
@@ -1,99 +1,73 @@
+                _setTagNameViewsEl: function (view, wantTagName) {
+                    if (view !== undefined && (view.$el.tagName === undefined || (wantTagName && wantTagName.isString()))) {
+                            view.$el.appendTo($(wantTagName));
+                            view.render();
```
рендеринг всех вью на заходе в приложение?
это лишнее
рендерить надо тогда, когда надо показать

madimp commented on public_html/js/router.js:
```diff
@@ -1,99 +1,73 @@
+                },
+
+                defaultAction: function () {
+                    Views.main.show();
```
как быть со скрытием предыдущих вью?

KirSlayk replied:
не совсем понял вопрос -- у меня же viewManager скрывает все

madimp commented on public_html/js/router.js:
```diff
@@ -1,99 +1,73 @@
+                },
+
+                logoutAction: function() {
+                    Views.logout.onSubmitEvent();
```
вообще странно выглядит
вызов обработчика события
очень странно
при этом это вью
есть мнение, что надо вызвать логаут у сессии

madimp commented on public_html/js/views/login.js:
```diff
@@ -2,35 +2,47 @@ define([
+            console.log("[views::login::render()]: called");
             console.log(this.$el);
             this.$el.html(this.template());
+            this.$el.on("submit", this._onSubmitEvent);
```
у вас на каджый рендер будет добавляться обработчик. сколько раз на логин приходили, столько раз и навесится
используйте view.events 
более того, контекст обработчика у вас будет не вью логина

madimp commented on public_html/js/views/logout.js:
```diff
@@ -0,0 +1,41 @@
+            var reqObj = {
+                "login": Session.getUser().login
+            };
+            Session.logout(reqObj);
```
сессия сама разве не может определить параметры разлогина?

madimp commented on public_html/js/views/logout.js:
```diff
@@ -0,0 +1,41 @@
+        },
+
+        render: function () {
+            console.log("[views::logout::render()]: called");
```
зачем вообще тут вью, которая невидима? что она отображает?

madimp commented on public_html/js/views/main.js:
```diff
@@ -2,45 +2,47 @@ define([
 
         initialize: function () {
+            $('#start').bind('clickStart', this._onSubmitEvent);
+            this.on('clickStart', this._onSubmitEvent);
```
что такое clickStart и почему на нем некий сабмит?

madimp commented on public_html/js/views/users_manager.js:
```diff
@@ -0,0 +1,29 @@
+
+        initialize: function(){
+            console.log("in UsersManager initialize");
+            this.usersCollection = new CollectionUsers();
```
зачем тут вью, если используется коллекция?

madimp commented on templates/login.xml:
```diff
@@ -3,52 +3,22 @@
-            <div class="panel-heading">
-                <div class="panel-title">Sign In</div>
+    <form class="center-block text-center">
+        <p><lable><input id="login-username" type="login"  class="form-control input-field center-block" name="username" value="" placeholder="email" required="required"></input></lable></p>
```
отформатируйте код

madimp commented on templates/login.xml:
```diff
@@ -3,52 +3,22 @@
     <style type="text/css">
         <fest:insert src="../public_html/css/login.css"/>
```
стили подключать инлайном странное решение. вынесете во внешний файл все стили

KirSlayk committed:
refactor code for pull request 30%

KirSlayk committed:
refactoring for pull-request 80%

FakeEmperor committed:
[WIP] Codebase in work

FakeEmperor committed:
[WIP] Forgot card jpeg

FakeEmperor committed:
[WIP] Commit it, dammit

FakeEmperor committed:
Merge branch 'master' of https://github.com/SlaykovskiyHDNANO/2016_1_Lime-n-Pineapple

FakeEmperor committed:
Huge update! Version is upped to 0.05a :smile:
[+] BaseView to simplify view creation and management.
[+] CSS Styles moved to SCSS and optimized
[-] Removed EventBus
[!] EventBus's role is taken by Backbone itself! See in sources (e.g.
views/login.js)
[+] Using .jsx extension to indicate ES6 language usage
[*] Changed inspection engine from JSLint to JSHint
[+] .babelrc and .jshintrc configs for Babel and JSHint respectively
Still in progress.

FakeEmperor committed:
Fix: Forgot to change version number (actually, not 0.05 but 0.1.5)

FakeEmperor committed:
Tests passed

FakeEmperor committed:
Added babel grunt tasks and session.js for tests

KirSlayk committed:
some changes

KirSlayk committed:
conflicts were solved

KirSlayk committed:
I AM CONFLICTS'S GOOOOOOD

FakeEmperor committed:
[WIP] New layouts; changed css transpiler to Compass; Changed logic on routing buttons

FakeEmperor committed:
fix: Removed from further commiting 'public_html/tmpl' folder and added it to .gitignore

KirSlayk committed:
trush

KirSlayk committed:
...

KirSlayk committed:
,,,

KirSlayk committed:
trush

KirSlayk committed:
снова поменял стили, не удаляйте пжл, я усталъ. снова ничо не едет

KirSlayk committed:
grunt-contrib-compass added

KirSlayk committed:
.

KirSlayk committed:
Merge branch 'master' into slaykV2

KirSlayk committed:
game start, low functional

KirSlayk committed:
some changes

KirSlayk committed:
added battle field

KirSlayk committed:
stupid AI

KirSlayk committed:
scale function

KirSlayk committed:
some

KirSlayk committed:
nothing work, reconctructing game

KirSlayk committed:
Recovery previos functional. Divided on moduls

KirSlayk committed:
modified functional and быдлокода стало меньше

KirSlayk committed:
primitive animation

KirSlayk committed:
-compiled.js to .js

KirSlayk committed:
defined modules and views for some class

KirSlayk committed:
some

KirSlayk committed:
infocard

KirSlayk committed:
infocard move is completed

KirSlayk committed:
change name of events total

KirSlayk committed:
filters added on containers

KirSlayk committed:
hitAreas for containers were fixed

KirSlayk committed:
disign improved

KirSlayk committed:
some changes olololo

KirSlayk committed:
more changes

KirSlayk committed:
ААААЛИИУЙЯЯЯЯ, battleInfoCard added sucsesfull

KirSlayk committed:
added central passed card id deck only

KirSlayk committed:
added EventsConfigModule

KirSlayk committed:
Scores on battle added

KirSlayk committed:
отступ

KirSlayk committed:
some change

KirSlayk commented:
Примите пулл, плиз @DmitryDorofeev

DmitryDorofeev commented:
ПР должен принимать ментор

KirSlayk commented:
@madimp примите, пожалуйста, пулл

madimp commented on public_html/js/engine/AbstractCardContainerModel.js:
```diff
@@ -0,0 +1,103 @@
+"use strict";
+
+var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticP ...
```
как это читать вообще?

madimp commented on public_html/js/engine/CardContainerModel.js:
```diff
@@ -0,0 +1,34 @@
+
+function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }
+
+function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.se ...
```
мне кажется, это все-таки какая-то лабуда

KirSlayk replied:
@madimp это скомпилированные jsx файлы. Они залились давно, не удалил просто с гита. Нужно читать jsx. ща удалю

DmitryDorofeev replied:
Это не тот jsx, о котором я подумал😒

KirSlayk committed:
deleted js files

KirSlayk committed:
deleted .js files AGAIN

madimp commented:
.sass-cache зачем в комитах?

KirSlayk committed:
.sass deleted

KirSlayk commented:
@madimp удалил все ненужное

madimp commented on public_html/js/engine/CardContainerModel.jsx:
```diff
@@ -0,0 +1,31 @@
+
+        }
+
+        //onClickBattleField(event, container, actionCard, infoCard) {
```
удалить

KirSlayk replied:
@madimp тут в скором времени будут лежать некоторые данные + уже все завязано на этом пустом (пока что) модуле

madimp replied:
я о коментариях

madimp commented on public_html/js/engine/abstract_player.jsx:
```diff
@@ -0,0 +1,206 @@
+                        this.cleanClickListenerForContainers();
+                        this.infoCard.alreadyGoingBack = !this.infoCard.alreadyGoingBack;
+                        this.infoCard.trigger(Events.Game.InfoCardModel.BackToDeck, cardModel);
+                        cardModel.trigger(Events.Game.AbstractCardModel.CleanClickEventCard);
```
абстрактный плеер подписывается конерктные на события конкретных моделей?
чего в нем абстрактного?

iketari commented on public_html/js/engine/InfoCardView.jsx:
```diff
@@ -0,0 +1,151 @@
+        './EventsConfig'
+    ],
+    function (Backbone, _, $, pixi, SETTING, Events) {
+        class InfoCardView{
```
Просто к сведению. JSX - где вы его откопали? Это ж какое-то старье. =)
Если есть желание кодить на JS "по-взрослому", очень советую посмотреть в сторону TypeScript.
Там правильные классы с protected, private, public, наследованием.
Типизация во всей свой красоте, с дженериками, интерфейсами, абстрактными классами и прочим.
Отличная поддержка нового стандарта.
Ну и как бонус шикарная поддержка IDE, много матреиалов в сети.
